src/ess/reflectometry/normalize.py

```diff
@@ -2,12 +2,83 @@
 # Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
 import scipp as sc
 
-from .types import IofQ, NormalizedIofQ, Reference, Sample
+from .types import (
+    CorrectionMatrix,
+    FootprintCorrectedData,
+    NormalizationFactor,
+    NormalizedIofQ,
+    QBins,
+    Sample,
+    WBins,
+)
+
+
+def normalization_factor(
+    da: FootprintCorrectedData[Sample],
+    corr: CorrectionMatrix,
+    qbins: QBins,
+    wbins: WBins,
+) -> NormalizationFactor:
+    '''The correction matrix gives us the expected intensity at each
+    (z_index, wavelength) bin assuming the reflectivity is one.
+    To normalize the sample measurement we need to integrate the total
```

Comment:
I still haven't fully understood why we need to do this, and why we couldn't compute the correction as a function of pixel and wavelength, get the sample data in the form of pixel vs wavelength, and then divide by the reference.
Only after that compute Q for the sample?

Reply:
For the sake of the discussion here let's say that the detector has two logical pixel dimensions $y$ and $z$.

If I understand your question correctly, instead of working with the intensity $I(z, \lambda) := \int I(y, z, \lambda) \ dy$ you would like to work directly with the intensity over (all) pixels and wavelength, $I(y, z, \lambda)$.

From this point we can proceed like in [my comment above](https://github.com/scipp/essreflectometry/pull/44#issuecomment-2095485927).

$I^{sample}(y, z, \lambda)$ is the sample intensity and $I^{reference}(y, z, \lambda)$ is the intensity from the reference measurement.

Our model for the sample intensity is:
$$I^{sample}(y, z, \lambda) = I^{reference}(y, z, \lambda) \ \hat{n}(Q^{reference}(y, z, \lambda))\  R(Q^{sample}(y, z, \lambda))$$
where $\hat{n}$ denotes supermirror efficiency and $R$ denotes sample reflectivity.

From what I understand the approach you are suggesting is to compute the reflectivity like this:
$$R(Q^{sample}(y, z, \lambda)) = \frac{I^{sample}(y, z, \lambda)}{I^{reference}(y, z, \lambda) \hat{n}(Q^{reference}(y, z, \lambda))} $$

and then histogram the reflectivities in every pixel/wavelength bin, that is $R(Q^{sample}(y, z, \lambda))$, in $Q^{sample}$. This is what you and I tried the other day when we went through the workflow together.

#### I think the reason why this doesn't work
 is that when we bin in an additional dimension $y$ we reduce the number of events in every bin and then we don't have enough statistics to estimate $I^{sample}(y, z, \lambda)$ and $I^{reference}(y, z, \lambda)$ sufficiently accurately.
The effect of this is that $R(Q^{sample}(y, z, \lambda))$ will be noisy.
It's only true that the mean of $R$ in every Q-bin is an unbiased estimator of $R$ in that Q-bin if the mean of the distribution of $R(Q^{sample}(y, z, \lambda))$ is equal to the mean of the distribution of $I^{sample}(y, z, \lambda)$ divided by the mean of the distribution $I^{reference}(y, z, \lambda)$. This is only true if we have sufficiently many counts in every $(y, z, \lambda)$ bin so that the Poisson distributed intensity measurements are approximately Binomial distributed.